### PR TITLE
Issue #969: Jobs with a hash in the name can't be opened

### PR DESCRIPTION
**Pull request by katelynienaber**
## Proposed changes

It's a fix for Issue #969. I didn't have trouble with the issue mentioned there. But the job output couldn't be opened, because function `vscode.Uri.parse` was interpreting the # incorrectly.

Fixed by using function `vscode.Uri.parse().with()` instead, to more explicitly parse the job info as an URI.

## Release Notes

Milestone: Backlog
Changelog: Fixed error when showing job output, for jobs that have a hash character (#) in the name.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

This was reported by at least two customers and should probably be merged ASAP.

### CODE REVIEW

**Comment by katelynienaber** [comment]
@zFernand0 Requesting your review, because there is a vulnerability reported?

In package `@zowe/cli`...maybe you know how to fix it?

**Comment by zFernand0** [comment]
After a quick update to the entire yarn.lock file, seems like the CLI vulnerabilities have been resolved.
Something we may want to consider is locking down the version of vscode in our devDependencies as this can cause issues when building a package.
For example, 
```typescript
public getLabel(): string {
    return this.label.trim(); 
    // This fails becuase `typeof this.label === 'string | TreeItemLabel'` and there is no `.trim()` on TreeItemLabel 
    // Instead we have to...
    return typeof this.label === "string" ? this.label.trim() : this.label?.label.trim();
}
```

**Review by jellypuno — COMMENTED** [review]
I know that the change that you have here is very simple but I don't know why you have 8 files edited. Is it because you executed prettify? Is this really a change or was it an accident?

File: .github/workflows/zowe-explorer-ftp-ci.yml [hunk 1]
```diff
@@ -43,14 +43,12 @@ jobs:
         env:
           CI: true
           NODE_OPTIONS: --max_old_space_size=4096
-
       # - name: Upload test results
```
[the remark this answers is not in the document]
Comment (chladnefazole) [reply .github/workflows/zowe-explorer-ftp-ci.yml:46]: I think it's just prettier, all the changes are just to whitespace (except for tests related to the spool provider).

**Review by lauren-li — APPROVED** [review]
This looks good to me! All unit and expected integration tests are passing, and I'm now able to open spool files from a job with a `#` character in the name.

Thanks @katelynienaber for this fix!

**Comment by jellypuno** [comment]
@katelynienaber This works for me as well however I am experiencing issues with Integration tests. I have errors in rename and create datasets

```
15) Extension Integration Tests
       Copying data sets
         Success Scenarios
           Member > Member
             "before each" hook for "Should copy a data set":
     TypeError: Cannot read property 'trim' of undefined

 11) Extension Integration Tests
       Enter Pattern
         should match data sets for multiple patterns:
     Error: Cannot resolve tree item for element 0/0:zcobol /0:No datasets found
  	at c:\Users\jp669971\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:86:61247
  	at processTicksAndRejections (internal/process/task_queues.js:97:5)
     
```

Is this also happening to you?

**Comment by katelynienaber** [comment]
@jellypuno those two tests are passing for me, maybe as well for @lauren-li ?
image

image

**Comment by jellypuno** [comment]
ok then so maybe it is just my set-up. thanks for checking!

**Review by jellypuno — APPROVED** [review]

**Comment by lauren-li** [comment]
> @jellypuno those two tests are passing for me, maybe as well for @lauren-li ?
> image
> 
> image

I had no issues with the copy data set test. The "match data sets for multiple patterns" test always times out for me, even on `master` branch, so I didn't note it as unusual when it did the same while I was testing this PR branch.

**Pull request merged**